motulator/common/model/_converter.py

```diff
@@ -11,14 +11,129 @@
+
+    Parameters
+    ----------
+    u_dc : float | callable
+        DC-bus voltage (V).
+    C_dc : float, optional
+        DC-bus capacitance (F). Default is None.
+    i_ext : callable, optional
+        External DC current, seen as disturbance, `i_ext(t)`. Default is zero,
+        ``lambda t: 0``.
+
+    """
+
+    def __init__(self, u_dc, C_dc=None, i_ext=lambda t: 0):
+        super().__init__()
+        self.i_ext = i_ext
+        self.par = SimpleNamespace(
+            u_dc=u_dc,
+            C_dc=C_dc,
+        )
+        # Initial values
+        self.u_dc0 = self.par.u_dc(0) if callable(
+            self.par.u_dc) else self.par.u_dc
+        # Only initialize states if dynamic DC model is used
+        if self.par.C_dc is not None:
+            self.state = SimpleNamespace(u_dc=self.u_dc0)
+            self.sol_states = SimpleNamespace(u_dc=[])
+        self.inp = SimpleNamespace(
+            u_dc=self.u_dc0,
+            i_ext=i_ext(0),
+            q_cs=None,
+            i_cs=0j,
+        )
+        self.sol_q_cs = []
+
+    @property
+    def u_dc(self):
+        """DC-bus voltage (V)."""
+        if self.par.C_dc is not None:
+            return self.state.u_dc.real
+        return self.inp.u_dc
+
+    @property
+    def u_cs(self):
+        """AC-side voltage (V)."""
+        return self.inp.q_cs*self.u_dc
+
+    @property
+    def i_dc(self):
+        """DC-side current (A)."""
+        return 1.5*np.real(self.inp.q_cs*np.conj(self.inp.i_cs))
+
+    def set_outputs(self, _):
+        """Set output variables."""
+        self.out.u_cs = self.u_cs
+        self.out.u_dc = self.u_dc
+
+    def set_inputs(self, t):
+        """Set input variables."""
+        self.inp.u_dc = self.par.u_dc(t) if callable(
+            self.par.u_dc) else self.par.u_dc
+        self.inp.i_ext = self.i_ext(t)
+
+    def rhs(self):
+        """
+        Compute the state derivatives.
+
+        Returns
+        -------
+        complex list, length 1
+            Time derivative of the complex state vector, [d_u_dc].
+
+        """
+        inp, par = self.inp, self.par
+        if par.C_dc is None:  # Check whether dynamic DC model is used
+            return None
+        d_u_dc = (inp.i_ext - self.i_dc)/par.C_dc
+        return [d_u_dc]
+
+    def meas_dc_voltage(self):
+        """
+        Measure the converter DC-bus voltage.
+
+        Returns
+        -------
+        u_dc : float
+            DC-bus voltage (V).
+
+        """
+        return self.u_dc
+
+    def post_process_states(self):
+        """Post-process data."""
+        data = self.data
+        if self.par.C_dc is None:
+            if callable(self.u_dc):
+                self.data.u_dc = self.u_dc(self.data.t)
+            else:
+                self.data.u_dc = np.full(np.size(self.data.t), self.u_dc)
+        else:
+            data.u_dc = data.u_dc.real
+        data.u_cs = data.q_cs*data.u_dc
+
+
+# %%
+class LegacyInverter(Subsystem):
```

Comment:
If possible, this LegacyInverter class could be avoided. It is fine to do the necessary modifications to the drive models. As we discussed, we can limit the diode bridge model to the case where its supply voltages are stiff and balanced.